use person ucr for migration and availing

calellowitz opened the pull request:
<!--- Provide a link to the ticket or document which prompted this change -->

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
This fixes https://dimagi-dev.atlassian.net/browse/ICDS-1113 by pulling from the person ucr, which does get updated, rather than the child health case ucr, which is not updated when a person is migrated (or no longer availing).

calellowitz committed:
use person ucr for migration and availing

emord reviewed (approved):
I think that you'll also want to fix ccs_record_monthly right? https://github.com/dimagi/commcare-hq/blob/c33cf1c1eadd5b725c53338892d928b1d83bc9e9/custom/icds_reports/utils/aggregation_helpers/distributed/ccs_record_monthly.py#L63 (though those are updated less frequently and likely less visible)

calellowitz committed:
Update person ucr fixture

update was done using the following sql
update "ucr_icds-cas_static-person_cases_v3_2ae0879a" p set migration_status=ch.is_migrated, registered_status=is_availing from "ucr_icds-cas_static-child_health_cases_a46c129f" ch where p.doc_id=ch.mother_id and p.supervisor_id=ch.supervisor_id;

calellowitz committed:
use person case migration and availing for ccs records

calellowitz committed:
use "IS DISTINCT FROM" to handle nulls

calellowitz committed:
update ccs record person cases

calellowitz committed:
update ccs output

calellowitz committed:
Fix agg table outputs

used same command as child health earlier in PR to backfill migration and availing data but fixed issues with conflicting ccs records sharing same person with following query
```
update "ucr_icds-cas_static-ccs_record_cases_cedcca39" p set person_case_id =new_id from (select doc_id, person_case_id || '_' || row_number as new_id from (select doc_id, person_case_id, is_migrated, is_availing, row_number() over w, count(*) over w as c  from "ucr_icds-cas_static-ccs_record_cases_cedcca39" window w as (partition by person_case_id, supervisor_id) )x  where x.c >1 and (is_migrated=1 or is_availing=0)) x where x.doc_id=p.doc_id;
```

calellowitz committed:
Refactor cas data export tests and fix outputs
